src/v2/coins/eth.ts

```diff
@@ -805,6 +812,182 @@ class Eth extends BaseCoin {
+      }
+
+      // Right now we only support 1 recipient
+      if (recipients.length !== 1) {
+        throw new Error('must send to exactly 1 recipient');
+      }
+      const recipientAddress = recipients[0].address;
+      const recipientAmount = recipients[0].amount;
+      const feeEstimateParams = {
+        recipient: recipientAddress,
+        hop: true,
+      };
+      const feeEstimate = yield this.feeEstimate(feeEstimateParams);
+      const gasLimit = feeEstimate.gasLimitEstimate;
+      // shift by 9 because result is in gwei and we want wei
+      const gasPrice = Math.round((feeEstimate.feeEstimate) / gasLimit);
+      const gasPriceMax = gasPrice * 5;
+      // Payment id as  current datetime so its always unique every time we build one
+      const paymentId = (new Date().toISOString());
+      const hopDigest: Buffer = Eth.getHopDigest([recipientAddress, recipientAmount, gasPriceMax, gasLimit, paymentId]);
+
+      const userReqSig = optionalDeps.ethUtil.addHexPrefix(secp256k1.sign(hopDigest, userPrvBuffer).signature.toString('hex'));
+
+      return {
+        hopParams: {
+          gasPriceMax,
+          userReqSig,
+          paymentId,
+        },
+        gasLimit,
+      };
+    }).call(this).asCallback(callback);
+  }
+
+  /**
+   * Validates that the hop prebuild from the HSM is valid and correct
+   * @param wallet The wallet that the prebuild is for
+   * @param hopPrebuild The prebuild to validate
+   * @param originalParams The original parameters passed to prebuildTransaction
+   * @param callback
+   * @returns void
+   * @throws Error if The prebuild is invalid
+   */
+  validateHopPrebuild(wallet: any, hopPrebuild: HopPrebuild, originalParams: any, callback: any): void {
+    return co(function *() {
+      const {
+        tx,
+        id,
+        signature,
+      } = hopPrebuild;
+
+      const {
+        recipients,
+      } = originalParams;
+
+      // first, validate the HSM signature
+      const bitgoKeychain: { pub: string } = yield this.keychains().get({ id: wallet.keyIds()[2] });
+      const bitgoPubkeyBuffer: Buffer = prova.HDNode.fromBase58(bitgoKeychain.pub).getPublicKeyBuffer();
+      const signatureBuffer: Buffer = Buffer.from(optionalDeps.ethUtil.stripHexPrefix(signature), 'hex');
+      const messageBuffer: Buffer = Buffer.from(optionalDeps.ethUtil.stripHexPrefix(id), 'hex');
+
+      const isValidSignature: boolean = secp256k1.verify(messageBuffer, signatureBuffer.slice(1), bitgoPubkeyBuffer);
+      if (!isValidSignature) {
+        throw new Error(`Hop txid signature invalid`);
+      }
+
+      // Then validate that the tx params actually equal the requested params
+      const originalAmount: string = recipients[0].amount;
+      const originalDestination: string = recipients[0].address;
+
+      const builtHopTx = new optionalDeps.EthTx(tx);
+      const hopAmount: string = BigNumber(optionalDeps.ethUtil.bufferToHex(builtHopTx.value)).toFixed(0);
```

Comment:
Nit: I'd rather you convert `originalAmount` to a `BigNumber` and compare that to the `hopAmount` instead of converting `hopAmount` to a string. We should try to compare numbers instead of strings where possible.

Reply:
done